### PR TITLE
Fix release-same-task-twice bug in fair balancer

### PR DESCRIPTION
**Pull request by schmichael**
Old code had a bug that could try to release the same task twice when
balancing. In practice it never hurt anything, but it was a fun source
of random test failures.

Please do #114 first!

### CODE REVIEW

File: balancer.go [hunk 1]
```diff
@@ -125,17 +125,22 @@ func (e *FairBalancer) Balance() []string {
...
 		return nil
 	}
 
+	releasetasks := make([]string, 0, shouldrelease)
+	releaseset := make(map[string]struct{}, shouldrelease)
+
 	random := rand.New(rand.NewSource(time.Now().UnixNano()))
 	nodetasks := e.bc.Tasks()
 	for len(releasetasks) < shouldrelease {
 		tid := nodetasks[random.Intn(len(nodetasks))].ID()
```
Comment (epsniff) [inline balancer.go]: I don't see that we Seed random.   Note this comment from goland rand:
```Random numbers are generated by a Source. Top-level functions, such as Float64 and Int, use a default shared Source that produces a deterministic sequence of values each time a program is run. Use the Seed function to initialize the default Source if different behavior is required for each run. The default Source is safe for concurrent use by multiple goroutines.```

Comment (epsniff) [inline balancer.go]: NM, I see it

**Comment by epsniff** [comment]
:+1:

**Comment by kyledj** [comment]
:checkered_flag::checkered_flag::checkered_flag:
:checkered_flag::no_entry_sign::checkered_flag:
:checkered_flag::checkered_flag::checkered_flag:

**Pull request merged**